signer/src/keypair.rs

```diff
@@ -73,10 +69,38 @@ impl Keypair {
         Ok(Keypair::from_parts_unsafe(secret, public))
     }
 
+    /// Deserialize a keypair from secret key scalar
+    pub fn from_secret(secret: ScalarField) -> Result<Self> {
+        let public = Self::derive_pubkey_curve_point(secret);
+        if !public.is_on_curve() {
+            return Err(KeypairError::NonCurvePoint);
+        }
+        // Safe now because we checked point is on the curve
+        Ok(Keypair::from_parts_unsafe(secret, public))
+    }
+
     /// Obtain the Mina address corresponding to the keypair's public key
     pub fn get_address(self) -> String {
         self.public.into_address()
     }
+
+    /// Validates the keypair
+    pub fn validate(&self) -> bool {
```

Comment:
it sounds like this check should happen automatically, not with a function that can be forgotten

Reply:
in mina-signer, one can construct an arbitrary keypair and validate that later, so it does require a separate API

e.g.

```javascript
const client = new Client({network: 'mainnet'})
const keypair = {privateKey: 'bla', publicKey: 'bla'}
expect(client.verifyKeypair(keypair)).toBe(true)
```

Reply:
It doable, but the goal is to create a wasm version of [mina-signer](https://www.npmjs.com/package/mina-signer) as a drop-in replacement, all APIs have to be compatible